targets/TARGET_STM/TARGET_STM32G4/STM32Cube_FW/CMSIS/stm32g431xx.h

```diff
@@ -9604,9 +9604,10 @@ typedef struct
 #define SYSCFG_MEMRMP_MEM_MODE_1        (0x2UL << SYSCFG_MEMRMP_MEM_MODE_Pos)  /*!< 0x00000002 */
 #define SYSCFG_MEMRMP_MEM_MODE_2        (0x4UL << SYSCFG_MEMRMP_MEM_MODE_Pos)  /*!< 0x00000004 */
 
-#define SYSCFG_MEMRMP_FB_MODE_Pos       (8U)
-#define SYSCFG_MEMRMP_FB_MODE_Msk       (0x1UL << SYSCFG_MEMRMP_FB_MODE_Pos)   /*!< 0x00000100 */
-#define SYSCFG_MEMRMP_FB_MODE           SYSCFG_MEMRMP_FB_MODE_Msk              /*!< User Flash Bank mode selection */
+// Commented out because G431KB only has one Flash Bank
+// #define SYSCFG_MEMRMP_FB_MODE_Pos       (8U)
+// #define SYSCFG_MEMRMP_FB_MODE_Msk       (0x1UL << SYSCFG_MEMRMP_FB_MODE_Pos)   /*!< 0x00000100 */
+// #define SYSCFG_MEMRMP_FB_MODE           SYSCFG_MEMRMP_FB_MODE_Msk              /*!< User Flash Bank mode selection */
```

Comment:
Sure we need to remove these lines ?

Reply:
I confirm that removing these changes doesn't stop compilation ?

Reply:
Yes - needs to be removed. If you don't remove the defines, the mbed-os will think that the device has 2 flash banks - and will try to swap them causing a crash.
Yes - removing these files don't stop compilation.

Sidenote:
It could be done in other ways, but I think this is the simplest/smallest way to do it since this "fix" is contained to the G431xx file only - other fixes means modifying more "global" files (flash_api.c, stm32g4xx_hal_flash.h) and adding more logic/defines. 

See: flash_api.c/GetBank => It uses SYSCFG_MEMRMP_FB_MODE to determine if the device has 1 or 2 flash banks.

Reply:
OK, so issue is in flash_api.c/GetBank, we need to correct it.

PS: FYI, files in TARGET_STM32G4/STM32Cube_FW are copied from https://github.com/STMicroelectronics/STM32CubeG4/tree/master/Drivers
So I try to avoid any patch :-)

Reply:
Yeah, I get that :)

It's a bit of a kludge fix that I've added, but it solves the problem in as few lines as possible :)

There's many other ways to fix it, but I think it requires feedback/decision from ST side on how to proceed.

Just thinking out loud, you could:

1. Add a #define somewhere (targets.json, stm32g431xx.h, ...) that states the number of flash banks (e.g. FLASH_BANK=2 or FLASH_BANK_SINGLE or ...).

2. Test for some 431xx #define instead of SYSCFG_MEMRMP in flash_api.c, but that can easily become kludgy if there's other devices in the G4 series that has only 1 flashbank.

3. Other ideas? :) (I'm not expert on mbed or ST specific code/architecture style)

The key issue is the coding style I guess, that the presence of a register (SYSCFG_MEMRMP) is used to determine the number of flash banks. I guess that would be an upstream "coding style" fix in ST32Cube_FW, which I guess is hard to implement practically/politically/decision wise. And maybe there are good legacy reasons for doing it that style. However, I did a quick search, and I did find the following post on the ST community that spots the same issue albeit with a different register #define: https://community.st.com/s/question/0D50X0000B32gtg/the-symbol-flashoptrdbank-is-used-as-an-option-and-a-register-bit-def

So I think we need some guidance here how to proceed.

Reply:
I will accept this a a quick workaround, because I didn't take time to update flash_api.c

Reply:
You should be able to remove these patches with #14149